sdk/spring/azure-identity-spring/src/main/java/com/azure/spring/identity/DefaultSpringCredentialBuilder.java

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.spring.identity;
+
+import com.azure.core.credential.TokenCredential;
+import com.azure.identity.ChainedTokenCredentialBuilder;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ *
+ */
+public class DefaultSpringCredentialBuilder extends SpringCredentialBuilderBase<DefaultSpringCredentialBuilder> {
+
+    /**
+     * Defines the AZURE_CREDENTIAL_PREFIX.
+     */
+    static final String AZURE_CREDENTIAL_PREFIX = "azure.credential.";
+
+    private String alternativePrefix;
+
+    public DefaultSpringCredentialBuilder alternativePrfix(String alternative) {
+        if (alternative != null) {
+            this.alternativePrefix = alternative + (alternative.endsWith(".") ? "" : ".");
+        }
+
+        return this;
+    }
+
+    public TokenCredential build() {
+        if (environment == null) {
+            throw new IllegalArgumentException("To build a spring credential the environment must be set");
+        }
+
+        List<TokenCredential> tokenCredentials = new ArrayList<>();
+
+        if (alternativePrefix != null) {
+            tokenCredentials.add(populateTokenCredential(alternativePrefix));
```

Comment:
If alternativePrefix is not null, then we will get managed identity credential if there is no such prefixed configuration in Environment. The order should be:

- Check alternativePrefix client-id, client-secret
- Check alternativePrefix client-id, client-certificate-path
- If there is only alternativePrefix client-id, try managed identity with client-id
- Check default prefix client-id, client-secret
- Check default prefix client-id, client-certificate-path
- If there is only default prefix client-id, try managed identity with client-id
- Use managed identity with no argument
-

Reply:
Changed.